lib/internal/test_runner/test.js

```diff
@@ -76,15 +75,41 @@ const { testNamePatterns, testOnlyFlag } = parseCommandLine();
 let kResistStopPropagation;
 
 function stopTest(timeout, signal) {
+  const deferred = createDeferredPromise();
+
   if (timeout === kDefaultTimeout) {
-    return once(signal, 'abort');
+    signal.addEventListener('abort', deferred.resolve, { __proto__: null, once: true });
+    deferred.cleanup = () => signal.removeEventListener('abort', deferred.resolve);
```

Comment:
i think this needs ObjectDefineProperty, otherwise a setter on Object.prototype for `cleanup` can intercept this?
```suggestion
    ObjectDefineProperty(deferred, 'cleanup', { configurable: true, writable: true, value: () => signal.removeEventListener('abort', deferred.resolve) });
```

Reply:
done